napari/_qt/qt_main_window.py

```diff
@@ -736,6 +755,8 @@ def __init__(self, viewer: 'Viewer', *, show: bool = True) -> None:
                 [self._qt_viewer.dockLayerControls.minimumHeight(), 10000],
                 Qt.Orientation.Vertical,
             )
+            # TODO: where to put this?
+            self._qt_window._warn_on_shimmed_plugins()
```

Comment:
I put the function on the `QtMainWindow` because that's who it's parented to, but I noticed to pop the dialog at the right time, it needs to be in `Window`. Should I be moving the function to `Window`? I wasn't sure, hence the TODO.

Reply:
@Czaki if I move it to `showEvent` [here](https://github.com/napari/napari/blob/main/napari/_qt/qt_main_window.py#L235), the dialog shows before the viewer opens. I put the call to `self._warn_on_shimmed_plugins` after the call to `super().showEvent`. I think that's a little strange, it would be nicer to have the viewer present in the background. Is there a way to achieve that?